shlib/shcrypto/hash.go

```diff
@@ -16,29 +16,35 @@ func keccak256(ds ...[]byte) []byte {
 	return state.Sum(h)
 }
 
-// HashBytesToBlock hashes the given byte slice and returns the result as a block.
-func HashBytesToBlock(ds ...[]byte) Block {
-	h := keccak256(ds...)
-	var b Block
-	copy(b[:], h)
-	return b
+func hashWithPrefix(p byte, b []byte) []byte {
+	return keccak256(append([]byte{p}, b...))
```

Comment:
I think you could save the initial array allocation of the `[]byte{p}`. That's because in the `append` a new array will be allocated anyways: capacity too small, so it will allocate a new array with `len(new) == len(b) + 1` 
```go
func hashWithPrefix(p byte, b []byte) []byte {
	a := make([]byte, len(b)+1)
	a[0] = p
	copy(a[1:], b)
	return keccak256(a)
}
```

Not sure if this is worth it and significant though